hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/FileTransferHandlerStreamVariable.java

```diff
@@ -91,18 +91,18 @@ public final OutputStream getOutputStream() {
             return ByteStreams.nullOutputStream();
         }
 
-        // we return the outputstream so we cannot close it here
+        // we return the output stream - we cannot close it here
         @SuppressWarnings("squid:S2095")
         final PipedOutputStream outputStream = new PipedOutputStream();
-        PipedInputStream inputStream = null;
+        final PipedInputStream inputStream;
         try {
             inputStream = new PipedInputStream(outputStream);
             publishUploadProgressEvent(fileUploadId, 0, fileSize);
             startTransferToRepositoryThread(inputStream, fileUploadId, mimeType);
         } catch (final IOException e) {
-            LOG.warn("Creating piped Stream failed {}.", e);
+            LOG.warn("Creating piped Stream failed {}.", e.getMessage());
             tryToCloseIOStream(outputStream);
-            tryToCloseIOStream(inputStream);
+            // input stream is not created in case of and IOException here
```

Comment:
are you completely sure about it, what if IOException happens in the middle of a transfer?

Reply:
yes - IOException is a checked exception and neither `publishUploadProgressEvent` nor `startTransferToRepositoryThread` do declare a `throws IOException` here.
Still... there could be other (runtime) exceptions (e.g. `RejectedExecutionException`) poping here up but also in the existing code the wouldn't have closed the stream. 😟